.github/workflows/dogfood.yml

```
@@ -0,0 +1,46 @@
+name: dogfood merge
+# This workflow is triggered on pushes to master and dogfood branches
+on:
+  pull_request:
+  push:
+  delete:
+    # There is no 'branch:' filter for the delete action,
+    # so we are using 'if:' inside the job instead.
+  workflow_dispatch:
+    # This allows to run the workflow manually from the Actions tab (can be useful for testing purposes)
+
+jobs:
+  dogfood_merge:
+    if: >-
+      (github.event_name == 'push' && (github.event.ref == 'refs/heads/master' || startsWith(github.event.ref, 'refs/heads/dogfood/')))
+      || (github.event_name == 'delete' && startsWith(github.event.ref, 'dogfood/'))
+    runs-on: ubuntu-latest-large
+    name: Update dogfood branch
+    permissions:
+      id-token: write # required for SonarSource/vault-action-wrapper
+    steps:
+      - name: get secrets
+        id: secrets
+        uses: SonarSource/vault-action-wrapper@d1c1ab4ca5ad07fd9cdfe1eff038a39673dfca64  # tag=2.4.2-1
+        with:
+          secrets: |
+            development/github/token/{REPO_OWNER_NAME_DASH}-coverage token | dogfood_token;
+            development/kv/data/slack token | slack_token;
```

```suggestion
            development/kv/data/slack webhook | slack_webhook;
```

It seems that this `Ilshidur/action-slack` requires a webhook (not just a token). Otherwise, it would [fail](https://github.com/SonarSource/rspec/actions/runs/15162883155/job/42633044558) with:

```
Message : connect ECONNREFUSED 127.0.0.1:80
```

This change was tested in https://github.com/SonarSource/rspec/compare/mp/rule-coverage...mj/test_dogfood_fail

---

@michael-jabbour-sonarsource good catch, thank you!